great_expectations/checkpoint/checkpoint.py

```diff
@@ -302,23 +299,7 @@ def _get_substituted_runtime_kwargs(
         return self._substitute_config_variables(config=substituted_config)
 
     def _substitute_config_variables(self, config: dict) -> dict:
-        substituted_config_variables = substitute_all_config_variables(
-            self.data_context.config_variables,
-            dict(os.environ),
-            self.data_context.DOLLAR_SIGN_ESCAPE_STRING,
-        )
-
-        substitutions = {
-            **substituted_config_variables,
-            **dict(os.environ),
-            **self.data_context.runtime_environment,
-        }
-
-        return substitute_all_config_variables(
-            data=config,
-            replace_variables_dict=substitutions,
-            dollar_sign_escape_string=self.data_context.DOLLAR_SIGN_ESCAPE_STRING,
-        )
+        return self.data_context._config_provider.substitute_config(config)
```

Comment:
Direct access to private attr feels bad - we have a few solutions here:
1. Keep it as is (this violation is done many times in this class)
2. Pass in config provider in the constructor
3. Add a property so we can access properly